neuralpredictors/layers/readouts/base.py

```diff
@@ -23,106 +23,34 @@ def initialize(self, *args, **kwargs):
         raise NotImplementedError("initialize is not implemented for ", self.__class__.__name__)
 
     def __repr__(self):
-        s = super().__repr__()
-        s += " [{} regularizers: ".format(self.__class__.__name__)
-        ret = []
-        for attr in filter(
-            lambda x: not x.startswith("_") and ("gamma" in x or "pool" in x or "positive" in x),
-            dir(self),
-        ):
-            ret.append("{} = {}".format(attr, getattr(self, attr)))
-        return s + "|".join(ret) + "]\n"
-
-
-class SpatialXFeatureLinear(nn.Module):
-    """
-    Factorized fully connected layer. Weights are a sum of outer products between a spatial filter and a feature vector.
-    """
-
-    def __init__(
-        self,
-        in_shape,
-        outdims,
-        bias,
-        normalize=True,
-        init_noise=1e-3,
-        constrain_pos=False,
-        **kwargs,
-    ):
-        super().__init__()
-        self.in_shape = in_shape
-        self.outdims = outdims
-        self.normalize = normalize
-        c, w, h = in_shape
-        self.spatial = Parameter(torch.Tensor(self.outdims, w, h))
-        self.features = Parameter(torch.Tensor(self.outdims, c))
-        self.init_noise = init_noise
-        self.constrain_pos = constrain_pos
-        if bias:
-            bias = Parameter(torch.Tensor(self.outdims))
-            self.register_parameter("bias", bias)
-        else:
-            self.register_parameter("bias", None)
-        self.initialize()
-
-    @property
-    def normalized_spatial(self):
-        if self.normalize:
-            norm = self.spatial.pow(2).sum(dim=1, keepdim=True)
-            norm = norm.sum(dim=2, keepdim=True).sqrt().expand_as(self.spatial) + 1e-6
-            weight = self.spatial / norm
+        return super().__repr__() + " [{}]\n".format(self.__class__.__name__)
+
+    def reduction_method(self, reduction="mean", average=None):
```

Comment:
It tends to be helpful for method name to describe an action. Perhaps call this something like `resolve_reduction_method`.